posts/2021/jupyter-based-diagnostics-overview.ipynb

```diff
@@ -0,0 +1,434 @@
+    "        - SiO2_FLUX_100m\n",
+    "        - ALK\n",
+    "        - DIC\n",
+    "        - O2\n",
+    "        - DOC\n",
+    "        - DOCr\n",
+    "        - NH4\n",
+    "        - NO3\n",
+    "        - SiO3\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up the \"Preprocessing\" Notebooks\n",
+    "\n",
+    "The first two bullet points from the \"traditional diagnostics\" workflow can be accomplished using a two primary notebooks, one for creating a data catalog (`00_Build_Catalog.ipynb`), the other accessing data from that catalog and applying a computation (`01_Compute_20yr_mean.ipynb`). A third notebook plots a visualization of the data catalog used in the analysis (`intro.ipynb`)\n",
+    "\n",
+    "If you are interested in how we built the data catalog, I encourage you check out the [\"Building Intake-ESM Catalogs from History Files\" blog post](https://ncar.github.io/esds/posts/2021/ecgtools-history-files-example/)\n",
+    "\n",
+    "If you are interested in how we computed the average over some time period using Dask, Xarray, and Intake-ESM, I encourage you check out our [\"Examining Diagnostics Using Intake-ESM and hvPlot\" blog post](https://ncar.github.io/esds/posts/2021/intake-esm-holoviews-diagnostics/)\n",
+    "\n",
+    "If you are interested in how we built the catalog visualization, I encourage you to check out the [\"Creating Model Documentation Using Jupyterbook and Intake-ESM\" blog post](https://ncar.github.io/esds/posts/2021/model_documentation_jupyterbook/)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using our Configuration File in the Notebooks\n",
+    "\n",
+    "At the top of every notebook, we call \n",
+    "\n",
+    "```python\n",
+    "from config import analysis_config\n",
+    "```\n",
+    "\n",
+    "which access the `analysis_config` variable from the `config.py` script:\n",
+    "\n",
+    "```python\n",
+    "import yaml\n",
+    "\n",
+    "with open(\"analysis_config.yml\", mode=\"r\") as fptr:\n",
+    "    analysis_config = yaml.safe_load(fptr)\n",
+    "\n",
+    "analysis_config['all_variables'] = analysis_config['variables']['physics'] + analysis_config['variables']['biogeochemistry']\n",
+    "analysis_config['all_cases'] = [analysis_config['reference_case_name']] + analysis_config['compare_case_name'] \n",
+    "```\n",
+    "\n",
+    "This allows us to access a dictionary with the configuration set in our `analysis_config.yml` file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to Parameterize Notebooks\n",
+    "An important requirement for a diagnostic package is that it be \"parameterizable\" such that a user can input a list of variables, and the package is able to \"inject\" those variables into the analysis. Here, we use the `variables` section of the `analysis_config.yaml` file:\n",
```

Comment:
```suggestion
    "An important requirement for a diagnostic package is that it be \"parameterizable\" such that a user can input a list of variables, and the package is able to \"inject\" those variables into the analysis. Here, we demonstrate a bare-minimum example of how this might work, specifying a list of `variables` in the `analysis_config.yaml` file:\n",
```